### DISCUSSION TITLE
cloudnative-pg installed with OLM (OPERATOR LIFECYCLE MANAGER) does not make backups

### DISCUSSION THREAD

**Discussion by Casraw** [cloudnative-pg/cloudnative-pg#414] [Q&A]
Hello,

i try to install the cloudnative-pg trough OLM (OPERATOR LIFECYCLE MANAGER). Later on we have more then one cloudnative-pg operator. For now it worked flaless. But when i try to make a backup. And create a backup manifest. I cannot describe the backup or see it with "kubectl get backup -A". But it is created with "kubectl create -f example-backup.yaml". And it is possible to delete it. Also the backup dont start at all.

Kubernetes: 1.22.6
OLM: 0.21.2
cloudnative-pg: 1.15.1

When more informations are needed i will provide everything i can :-). Thank you

**Comment by jlong49** [comment]
Hey @Casraw, can you share the logs of the cluster and the Operator when you took the backup?

**Comment by Casraw** [comment]
Sure i can. And thank you :-)

Operator:
`{"level":"info","ts":1657700610.545018,"logger":"backup-resource","msg":"validate create","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"error","ts":1657701012.1237917,"msg":"executing backup","controller":"pooler","controllerGroup":"postgresql.k8s.enterprisedb.io","controllerKind":"Pooler","backup":{"name":"backup-example","namespace":"mr5335"},"namespace":"mr5335","name":"backup-example","reconcileID":"da9aa9da-3442-49ce-8b07-ecb7a18c766c","uuid":"c05925c6-0284-11ed-a8e7-8e629c98c26d","stdout":"","stderr":"{\"level\":\"info\",\"ts\":1657701012.1190476,\"msg\":\"Error while requesting backup\",\"logging_pod\":\"example-postgres-cluster-1\",\"backupURL\":\"http://localhost:8010/pg/backup\",\"statusCode\":500,\"body\":\"error while getting backup: backups.postgresql.k8s.enterprisedb.io \\\"backup-example\\\" not found\\n\"}\nError: invalid status code: 500\n","error":"command terminated with exit code 1 - {\"level\":\"info\",\"ts\":1657701012.1190476,\"msg\":\"Error while requesting backup\",\"logging_pod\":\"example-postgres-cluster-1\",\"backupURL\":\"http://localhost:8010/pg/backup\",\"statusCode\":500,\"body\":\"error while getting backup: backups.postgresql.k8s.enterprisedb.io \\\"backup-example\\\" not found\\n\"}\nError: invalid status code: 500\n","stacktrace":"github.com/EnterpriseDB/cloud-native-postgres/controllers.StartBackup\n\tcontrollers/backup_controller.go:225\ngithub.com/EnterpriseDB/cloud-native-postgres/controllers.(*BackupReconciler).Reconcile\n\tcontrollers/backup_controller.go:179\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}
{"level":"error","ts":1657701012.493322,"msg":"Reconciler error","controller":"pooler","controllerGroup":"postgresql.k8s.enterprisedb.io","controllerKind":"Pooler","backup":{"name":"backup-example","namespace":"mr5335"},"namespace":"mr5335","name":"backup-example","reconcileID":"da9aa9da-3442-49ce-8b07-ecb7a18c766c","error":"Backup.postgresql.k8s.enterprisedb.io \"backup-example\" not found","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}
{"level":"info","ts":1657705272.3234007,"logger":"backup-resource","msg":"default","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"info","ts":1657705272.3346252,"logger":"backup-resource","msg":"validate create","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"info","ts":1657705272.3466587,"msg":"Starting backup","controller":"pooler","controllerGroup":"postgresql.k8s.enterprisedb.io","controllerKind":"Pooler","backup":{"name":"backup-example","namespace":"mr5335"},"namespace":"mr5335","name":"backup-example","reconcileID":"7dd03484-fc54-4dd9-b1eb-60587a61aa30","uuid":"ee828b13-028f-11ed-a8e7-8e629c98c26d","cluster":"example-postgres-cluster","pod":"example-postgres-cluster-1"}
{"level":"info","ts":1657705300.980763,"logger":"backup-resource","msg":"default","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"info","ts":1657705300.9855752,"logger":"backup-resource","msg":"validate create","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"info","ts":1657705668.8378494,"logger":"backup-resource","msg":"default","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"info","ts":1657705668.84803,"logger":"backup-resource","msg":"validate create","version":"v1","name":"backup-example","namespace":"mr5335"}
{"level":"error","ts":1657705814.0108862,"msg":"executing backup","controller":"pooler","controllerGroup":"postgresql.k8s.enterprisedb.io","controllerKind":"Pooler","backup":{"name":"backup-example","namespace":"mr5335"},"namespace":"mr5335","name":"backup-example","reconcileID":"7dd03484-fc54-4dd9-b1eb-60587a61aa30","uuid":"ee828b13-028f-11ed-a8e7-8e629c98c26d","stdout":"","stderr":"{\"level\":\"info\",\"ts\":1657705814.006315,\"msg\":\"Error while requesting backup\",\"logging_pod\":\"example-postgres-cluster-1\",\"backupURL\":\"http://localhost:8010/pg/backup\",\"statusCode\":500,\"body\":\"error while getting backup: backups.postgresql.k8s.enterprisedb.io \\\"backup-example\\\" not found\\n\"}\nError: invalid status code: 500\n","error":"command terminated with exit code 1 - {\"level\":\"info\",\"ts\":1657705814.006315,\"msg\":\"Error while requesting backup\",\"logging_pod\":\"example-postgres-cluster-1\",\"backupURL\":\"http://localhost:8010/pg/backup\",\"statusCode\":500,\"body\":\"error while getting backup: backups.postgresql.k8s.enterprisedb.io \\\"backup-example\\\" not found\\n\"}\nError: invalid status code: 500\n","stacktrace":"github.com/EnterpriseDB/cloud-native-postgres/controllers.StartBackup\n\tcontrollers/backup_controller.go:225\ngithub.com/EnterpriseDB/cloud-native-postgres/controllers.(*BackupReconciler).Reconcile\n\tcontrollers/backup_controller.go:179\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}
{"level":"error","ts":1657705814.3493,"msg":"Reconciler error","controller":"pooler","controllerGroup":"postgresql.k8s.enterprisedb.io","controllerKind":"Pooler","backup":{"name":"backup-example","namespace":"mr5335"},"namespace":"mr5335","name":"backup-example","reconcileID":"7dd03484-fc54-4dd9-b1eb-60587a61aa30","error":"Backup.postgresql.k8s.enterprisedb.io \"backup-example\" not found","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}
`
Postgres:
Did not generate any logs.

**Comment by sxd** [comment]
I'm turning this into a discussion since doesn't look like an issue yet, if an issue with the code is found then an issue with the proper information can be created
